public/utils/helpers.tsx

```diff
@@ -0,0 +1,166 @@
+
+export const parseStringsToOptions = (strings: string[]) => {
+  return strings.map((str) => ({ id: str, label: str }));
+};
+
+export const renderTime = (time: number) => {
+  const momentTime = moment(time);
+  if (time && momentTime.isValid()) return momentTime.format('MM/DD/YY h:mm a');
+  return DEFAULT_EMPTY_DATA;
+};
+
+export function createTextDetailsGroup(
+  data: { label: string; content: any; url?: string }[],
+  columnNum?: number
+) {
+  const createFormRow = (label: string, content: string, url?: string) => (
+    <EuiFormRow label={<EuiText color={'subdued'}>{label}</EuiText>}>
+      {url ? (
+        <EuiLink>{content ?? DEFAULT_EMPTY_DATA}</EuiLink>
+      ) : (
+        <EuiText>{content ?? DEFAULT_EMPTY_DATA}</EuiText>
+      )}
+    </EuiFormRow>
+  );
+  return data.length <= 1 ? (
+    !data.length ? null : (
+      createFormRow(data[0].label, data[0].content, data[0].url)
+    )
+  ) : (
+    <>
+      <EuiFlexGroup>
+        {data.map(({ label, content, url }) => {
+          return (
+            <EuiFlexItem
+              key={label}
+              grow={false}
+              style={{ minWidth: `${100 / (columnNum || data.length)}%` }}
+            >
+              {createFormRow(label, content, url)}
+            </EuiFlexItem>
+          );
+        })}
+      </EuiFlexGroup>
+      <EuiSpacer size={'xl'} />
+    </>
+  );
+}
+
+export function parseSchedule({ period: { interval, unit } }: PeriodSchedule) {
+  return `Every ${interval} ${unit.toLowerCase()}`;
+}
+
+export function ruleItemInfosToItems(
+  detectorType: string,
+  ruleItemsInfo: RuleItemInfo[]
+): RuleItem[] {
+  if (ruleItemsInfo) {
+    return ruleItemsInfo.map((itemInfo) => ({
+      id: itemInfo._id,
+      active: itemInfo.enabled,
+      description: itemInfo._source.description,
+      library: itemInfo.prePackaged ? 'Sigma' : 'Custom',
+      logType: detectorType.toLowerCase(),
+      name: itemInfo._source.title,
+      severity: itemInfo._source.level,
+    }));
+  }
+
+  return [];
+}
+
+export function getUpdatedEnabledRuleIds(
+  existingEnabledIds: Set<string>,
+  ruleId: string,
+  isActive: boolean
+) {
+  let newEnabledIds;
+  // 1. not enabled previously
+  const wasActive = existingEnabledIds.has(ruleId);
+  if (wasActive && !isActive) {
+    const clonedIds = new Set(existingEnabledIds);
+    clonedIds.delete(ruleId);
+    newEnabledIds = [...clonedIds];
+  }
+  // 2. enabled previously and now disabled
+  else if (!wasActive && isActive) {
+    const clonedIds = new Set(existingEnabledIds);
+    clonedIds.add(ruleId);
+    newEnabledIds = [...clonedIds];
+  }
+
+  return newEnabledIds;
+}
+
+export function renderVisualization(spec: TopLevelSpec, containerId: string) {
+  let view;
+
+  try {
+    renderVegaSpec(compile({ ...spec, width: 'container', height: 400 }).spec).catch((err: Error) =>
+      console.error(err)
+    );
+  } catch (error) {
+    console.log(error);
```

Comment:
Not blocking for this release, but would `console.warn` of `console.error` make more sense here?

Reply:
Will replace this with a toast instead

Reply:
sending fix in follow up PR where all toasts will be added